### PR TITLE
Fix click behavior for links

### LINKED ISSUE #24756: Click Behavior drops FullApp overrides
**Describe the bug**
The Click Behavior uses `clickLink`, which essentially does a `window.location` change, which are much slower than regular `history.push` used elsewhere, but it also causes the [new overrides](https://www.metabase.com/docs/latest/embedding/full-app-embedding.html#overriding-selected-full-app-components-with-parameters) like `top_nav=true` to be lost.

**To Reproduce**
1. Create a dashboard with a card that has Click Behavior to another dashboard (regular Dashboard, not Custom URL)
2. View via FullApp embedding, where initial iframe src has overrides (`.../?top_nav=true`)
image
3. Click the card to go to the other dashboard, after browser load, it will now show the other dashboard, but without the overrides
image

**Expected behavior**
It would be great if Click Behaviors used history.push, which would be much quicker, but I remember there were some reasons for using window.location a long time ago. That would solve this problem.
Alternatively, remember overrides in browser session.
Or parse them on via Click Behavior. Though might be tricky to parse-thru for Custom URL, since we would have to detect if the URL is inside of Metabase, but we could control it for Dashboard and Question links.

**Information about your Metabase Installation:**
1.44.0

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/24756
Reproduces https://github.com/metabase/metabase/pull/24762

How to test:
- Set `IFRAMED = true` here https://github.com/metabase/metabase/blob/4ac2ef97082c0a34c71aef7f8be550910ab828bc/frontend/src/metabase/lib/dom.js#L13
- Open Metabase with a full-app embedding param, e.g. `http://localhost:3000?top_nav=true`
- Open a dashboard with a card 
- Add a click behavior with a link that navigates to a place within the app to a card
- Click on the card
- Make sure the navigation happens and the top nav bar is preserved

### CODE REVIEW

**Review by alxnddr — COMMENTED** [review]
It breaks the case:
- Create a dashboard with a filter
- Enter any filter value and copy the url with the filter parameter
- On the same dashboard, add a link click behavior with the url from the previous step
- Try to use the click behavior — it does not work

File: frontend/src/metabase/lib/dom.js [hunk 1]
```diff
@@ -361,6 +366,16 @@ const getOrigin = url => {
   }
 };
 
+const getLocation = url => {
```
Comment (ranquild) [inline frontend/src/metabase/lib/dom.js:369]: matches `react-router`, they also use `querystring`

File: frontend/src/metabase/visualizations/lib/action.js [hunk 2]
```diff
@@ -15,7 +16,13 @@ export function performAction(action, { dispatch, onChangeCardAndRun }) {
     const url = action.url();
     const ignoreSiteUrl = action.ignoreSiteUrl;
     if (url) {
-      open(url, { ignoreSiteUrl });
+      open(url, {
+        openInSameOrigin: (url, location) => {
+          dispatch(push(location));
+          dispatch(setParameterValuesFromQueryParams(location.query));
```
Comment (ranquild) [inline frontend/src/metabase/visualizations/lib/action.js:22]: There seems to be no clean way to implement the fix, but this approach seems to be the most straightforward.

**Comment by ranquild** [comment]
> It breaks the case:
> 
> * Create a dashboard with a filter
> * Enter any filter value and copy the url with the filter parameter
> * On the same dashboard, add a link click behavior with the url from the previous step
> * Try to use the click behavior — it does not work

Fixed, please take a look

File: frontend/src/metabase/visualizations/lib/action.js (hunk 2 above)
Comment (WiNloSt) [reply frontend/src/metabase/visualizations/lib/action.js:22]: I'm not entirely sure if this line has anything to do with the fix. Could you elaborate on this? I tried removing this line and the bug still seems fixed.

I think it has something to do with dashboard parameters, but I'm not totally sure.

Comment (ranquild) [reply frontend/src/metabase/visualizations/lib/action.js:22]: Yes, that's it

**Review by alxnddr — APPROVED** [review]
👍

**Review by WiNloSt — APPROVED** [review]
Looks good to me 👍

**Comment by ranquild** [comment]
I'm not confident enough in this fix to backport it.

**Pull request merged**